craft_parts/plugins/poetry_plugin.py

```diff
@@ -0,0 +1,126 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2020-2024 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The python plugin."""
+
+import shlex
+import shutil
+import subprocess
+from typing import Literal
+
+import pydantic
+from overrides import override
+
+from craft_parts import errors
+from craft_parts.plugins import validator
+
+from .base import BasePythonPlugin
+from .properties import PluginProperties
+
+
+class PoetryPluginProperties(PluginProperties, frozen=True):
+    """The part properties used by the python plugin."""
+
+    plugin: Literal["poetry"] = "poetry"
+
+    poetry_with: set[str] = pydantic.Field(
+        default_factory=set,
+        title="Optional dependency groups",
+        description="optional dependency groups to include when installing.",
+    )
+
+    # part properties required by the plugin
+    source: str  # pyright: ignore[reportGeneralTypeIssues]
+
+
+class PoetryPluginEnvironmentValidator(validator.PluginEnvironmentValidator):
+    """Check the execution environment for the Rust plugin.
+
+    :param part_name: The part whose build environment is being validated.
+    :param env: A string containing the build step environment setup.
+    """
+
+    _options: PoetryPluginProperties
+
+    @override
+    def validate_environment(
+        self, *, part_dependencies: list[str] | None = None
+    ) -> None:
+        """Ensure the environment has the dependencies to build Rust applications.
+
+        :param part_dependencies: A list of the parts this part depends on.
+        """
+        if shutil.which("python3") is None:
```

Comment:
To create a venv it needs a python executable, but I guess technically we don't need a `python3` on `$PATH`

Reply:
I'm thinking specifically about `PARTS_PYTHON_INTERPRETER` here. For example, there's a spread test for python 3.6 in rockcraft that relies on `PARTS_PYTHON_INTERPRETER` because apparently the python3- symlink deal doesn't work with python 3.6 (I don't remember the details)
https://github.com/canonical/rockcraft/commit/08e6a28c42624895d47dcd3208cd33d0fb8e220e#diff-586d5a5b9ea7f941ad7e0d6cf6f5b874fb56dbed7401733c26991521fa520cb8

Reply:
Yeah, good point. I've removed that bit.